src/cloud/validateCode.js

```diff
@@ -60,52 +59,60 @@ const setUserStatus = async (user, reservation = null) => {
     if (reservation) {
       user.set('status', 'inactive');
     } else {
-      // TODO: Uncomment when we use again the currentQuePosition logic.
-      // user.set('quePosition', currentQuePosition);
-      // if (maxQuePosition >= currentQuePosition) {
-      //   user.set('status', 'inactive');
-      // } else {
-      //   user.set('status', 'waitlist');
-      // }
+      const hasWaitListConversation = await ChatService.getConversationByCid(
+        `messaging:${user.id}_waitlist_conversation`,
+      );
+
       user.set('status', 'waitlist');
-      // TODO: Uncomment when the app (frontend) is ready to use it.
-      // await ChatService.createConversation(
-      //   user,
-      //   `${user.id}_invitation_conversation_${new Date().getTime()}`,
-      //   'invitation'
-      // );
-
-      const onboardingRole = await new Parse.Query(Parse.Role)
-        .equalTo('name', ONBOARDING_ADMIN)
-        .first();
-      if (onboardingRole) {
-        // If the role is defined, get the first user with it
-        const admin = await onboardingRole.get('users').query().first();
-        if (admin) {
-          await ChatService.createConversation(
-            user,
-            `${user.id}_waitlist_conversation`,
-            'messaging',
-            'Benji, Co-Founder',
-            [admin.id, user.id]
+
+      if (!hasWaitListConversation) {
+        // TODO: Uncomment when we use again the currentQuePosition logic.
+        // user.set('quePosition', currentQuePosition);
+        // if (maxQuePosition >= currentQuePosition) {
+        //   user.set('status', 'inactive');
+        // } else {
+        //   user.set('status', 'waitlist');
+        // }
+      
+        // TODO: Uncomment when the app (frontend) is ready to use it.
+        // await ChatService.createConversation(
+        //   user,
+        //   `${user.id}_invitation_conversation_${new Date().getTime()}`,
+        //   'invitation'
+        // );
+
+        // Retrieve the user with the phoneNumber
+        const userQuery = new Parse.Query(Parse.User);
+        userQuery.equalTo('phoneNumber', BENJI_PHONE_NUMBER);
+
+        const admin = await userQuery.first({ useMasterKey: true });
+
+        await ChatService.createConversation(
```

Comment:
Add a check to see if a waitlist conversation has already been created for this user. They could login, delete app, come back in and still be on the waitlist, and then would have 2 waitlist conversations